api/agent/agent.go

```diff
@@ -487,6 +488,16 @@ func (a *agent) launch(ctx context.Context, slots chan<- slot, call *call, tok R
 	ch := make(chan error, 1)
 
 	if !protocol.IsStreamable(protocol.Protocol(call.Format)) {
+
+		deadline, err := common.GetRemainingTime(ctx)
```

Comment:
this seems more like a `panic` to me as it means that somebody removed the timeout surrounding the call invocation (very bad). this env var is build-able in the early, shared stages of `getSlot` and we could do it before we have a handle on a token slot so that we don't have to be worried about leakage.

Reply:
will also cover hot functions, which we should add this to, as well

Reply:
ah, i see, hot is handled in `Start` separately. headers are a mess (i table flipped and gave up for now). wdyt about just setting both the env var and the header, for now, in the start of `getSlot` ? then it's in one place and we can move it around more easily and no handles on slots or anything. i still think it's a panic-able offense to not find a deadline, but it seems like we should just expect one or not set the env var if deadline is not there (instead of error/panic)